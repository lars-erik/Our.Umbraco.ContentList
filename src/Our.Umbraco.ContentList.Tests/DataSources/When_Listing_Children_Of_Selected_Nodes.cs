using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;
using Our.Umbraco.ContentList.DataSources;
using Our.Umbraco.ContentList.Models;
using Our.Umbraco.ContentList.Parameters;
using Our.Umbraco.ContentList.Tests.Support;
using Our.Umbraco.ContentList.Web.Models;
using Umbraco.Cms.Core.PublishedCache;

namespace Our.Umbraco.ContentList.Tests.DataSources
{
    [TestFixture]
    class When_Listing_Children_Of_Selected_Nodes
    {
        private UmbracoSupport support;

        [SetUp]
        public void Setup()
        {
            var cacheSupport = new ContentCacheSupport();
            var tree = cacheSupport.GetFromJsonResource(GetType().FullName);
            support = new UmbracoSupport(tree, Tests.Setup.ContentTypes);
            support.Setup();
        }

        [TearDown]
        public void TearDown()
        {
            support.TearDownUmbraco();
        }

        [Test]
        public void All_Published_Children_Are_Listed()
        {
            var ctx = support.GetUmbracoContext();
            var home = ctx.Content.GetById(1000);

            var dataSource = new ChildrenOfMultipleDataSource(support.GetService<IPublishedSnapshotAccessor>());
            var result = dataSource.Query(
                new ContentListQuery
                {
                    ContextContent = home,
                    CustomParameters =
                    {
                        { ListableSorting.Parameter.Key, ListableSorting.Parameter.Config.Items[0].Value },
                        { "nodes", "1001,1002" }
                    }
                },
                new QueryPaging(10)
            );

            Assert.That(result.ToList(), Has.Count.EqualTo(4));
            Assert.That(result.First(), Is.TypeOf<Page>());

            Approvals.VerifyJson(result.ToJson());

        }

    }
}
