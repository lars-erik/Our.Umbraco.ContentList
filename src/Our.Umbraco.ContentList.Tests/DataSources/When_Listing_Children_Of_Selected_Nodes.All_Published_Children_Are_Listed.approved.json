[
    {
        "ContentTypeName": "page",
        "ListImageUrl": null,
        "Url": "/item-1/item-11/",
        "SortDate": "0001-01-01T00:00:00",
        "Layout": null,
        "ListHeading": "Item 1.1",
        "ListImage": null,
        "ListSummary": {},
        "ReadMoreText": null,
        "Name": "Item 1.1",
        "UrlSegment": "item-11",
        "SortOrder": 0
    },
    {
        "ContentTypeName": "page",
        "ListImageUrl": null,
        "Url": "/item-1/item-12/",
        "SortDate": "0001-01-01T00:00:00",
        "Layout": null,
        "ListHeading": "Item 1.2",
        "ListImage": null,
        "ListSummary": {},
        "ReadMoreText": null,
        "Name": "Item 1.2",
        "UrlSegment": "item-12",
        "SortOrder": 1
    },
    {
        "ContentTypeName": "page",
        "ListImageUrl": null,
        "Url": "/item-2/item-21/",
        "SortDate": "0001-01-01T00:00:00",
        "Layout": null,
        "ListHeading": "Item 2.1",
        "ListImage": null,
        "ListSummary": {},
        "ReadMoreText": null,
        "Name": "Item 2.1",
        "UrlSegment": "item-21",
        "SortOrder": 0
    },
    {
        "ContentTypeName": "page",
        "ListImageUrl": null,
        "Url": "/item-2/item-22/",
        "SortDate": "0001-01-01T00:00:00",
        "Layout": null,
        "ListHeading": "Item 2.2",
        "ListImage": null,
        "ListSummary": {},
        "ReadMoreText": null,
        "Name": "Item 2.2",
        "UrlSegment": "item-22",
        "SortOrder": 1
    }
]